UI/obs-app.cpp

```diff
@@ -1180,6 +1189,8 @@ void OBSApp::AppInit()
 
 	if (!MakeUserDirs())
 		throw "Failed to create required user directories";
+	if (!MakeRecordingDir())
+		throw "Failed to create recording directory";
```

Comment:
This should be unnecessary as the directory is created on demand when a recording is started. OBS should not create an empty folder in the user's video folder just from being run.

Reply:
1. It does not create the folder on demand in my testing
![image](https://github.com/user-attachments/assets/c3409bff-9370-42d3-a2a5-d959538ebdf6)

2. I disagree and believe the folder should be initialized when OBS first runs like any app other storage location

Reply:
This isn't an app storage location though, this is the user's data folders. Well-designed apps should respect a user's folder layout and not create empty files and directories.

Reply:
Do you have some sort of guidance or documentation from Microsoft about that?

I have multiple applications on my system that have created folders for themselves in the Videos library location

Reply:
It's just good behavior, just because other apps are shitty doesn't mean we should also be shitty. There's [plenty of evidence of of users who desire such behavior](https://www.google.com/search?q=prevent+apps+from+creating+folders+in+my+documents) from their apps.

Reply:
One's shitty behaviour is another's expected behaviour though. And by that token I'd rather have an application create files in its own managed subdirectory than mess up the root of a data storage location.

I do agree that the directory should be created upon first use though. I don't remember Shadowplay creating directories for recordings or highlights until I actually triggered either.

Reply:
I concur with first use, as a user who does not enjoy all the random empty folders I have no intention of using.